src/main/java/com/example/validation/FieldMatchValidation.java

```diff
@@ -0,0 +1,30 @@
+package com.example.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.apache.commons.beanutils.BeanUtils;
+
+public class FieldMatchValidation implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstFieldName = constraintAnnotation.firstField();
+        secondFieldName = constraintAnnotation.secondField();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
+        try {
```

Comment:
Or you can use Spring BeanWrapperImpl instead of BeanUtils
```suggestion
        Object field = new BeanWrapperImpl(value).getPropertyValue(this.firstFieldName);
        Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.secondFieldName);
        return Objects.equals(field, fieldMatch);
```